common-libs/user-management/src/UserManagementService.ts

```diff
@@ -177,17 +194,25 @@ export class UserManagementService {
 
   async completeSignUpForEmailOrPhone(token: string, confirmationCode: string) {
     const { login, shortPassword } = this.parseSignUpToken(token)
-    this._loginShouldBeEmailOrPhoneNumber(login)
     await this._completeSignUp(login, confirmationCode)
     const cognitoTokens = await this._logInWithPassword(login, shortPassword, true)
     return { cognitoTokens, shortPassword }
   }
 
-  async doesUnconfirmedUserExist(username: string) {
-    const normalizedUsername = normalizeUsername(username)
-    return this._cognitoIdentityService.doesUnconfirmedUserExist(normalizedUsername)
+  /**
+   * Uses an ep on KeysService to check if record exists or not for given username or alias (email / phone)
+   * @param login
+   */
+  async doesUnconfirmedUserExist(login: string) {
+    const { userExists, isUnconfirmed } = await this.doesUserExist(login)
```

Comment:
@anton-iskryzhytskyi, how we know that user is unconfirmed? As i understand, we query users by attributes but do we have a way to check the confirmation status?
We could check attributes or do a doesConfirmedUser exist call in pair with this method

Reply:
to answer on your question I'd like to send you to the implementation of AWB there
`const isUnconfirmed = userExists && !(user.UserStatus === 'CONFIRMED')`

Reply:
oki got it